tests/foreman/maintain/test_service.py

```diff
@@ -0,0 +1,309 @@
+"""Test class for satellite-maintain upgrade functionality
+
+:Requirement: foreman-maintain
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: ForemanMaintain
+
+:Assignee: gtalreja
+
+:TestType: Functional
+
+:CaseImportance: Critical
+
+:Upstream: No
+"""
+import pytest
+from fauxfactory import gen_string
+
+from robottelo.config import settings
+from robottelo.constants import HAMMER_CONFIG
+from robottelo.constants import maintain_hammer_yml
+from robottelo.constants import satellite_answer_file
+from robottelo.logging import logger
+from robottelo.utils.issue_handlers import is_open
+
+
+@pytest.fixture
+def missing_hammer_config(request, sat_maintain):
+    """Setup/teardown fixture used by test_positive_service_restart_without_hammer_config"""
+    flag = 'initial_admin_password'
+    sat_maintain.execute(
+        f'sed -i "s/{flag}: {settings.server.admin_password}/{flag}: invalid/g" '
+        f'{satellite_answer_file}'
+    )
+    sat_maintain.execute(f'mv {HAMMER_CONFIG} /tmp/foreman.yml')
+    sat_maintain.execute(f'mv {maintain_hammer_yml} /tmp/foreman-maintain-hammer.yml')
+
+    @request.addfinalizer
+    def _finalize():
+        sat_maintain.execute(
+            f'sed -i "s/{flag}: invalid/{flag}: {settings.server.admin_password}/g" '
+            f'{satellite_answer_file}'
+        )
+        sat_maintain.execute(f'mv /tmp/foreman.yml {HAMMER_CONFIG}')
+        sat_maintain.execute(f'mv /tmp/foreman-maintain-hammer.yml {maintain_hammer_yml}')
+
+
+@pytest.mark.include_capsule
+def test_positive_service_restart(sat_maintain):
+    """Restart services using satellite-maintain service subcommand
+
+    :id: c5a38994-8c14-40c7-bc6a-1cfc68fc2d28
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain service restart
+
+    :expectedresults: Satellite services should restart.
+    """
+    result = sat_maintain.cli.Service.restart()
+    logger.info(result.stdout)
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+
+@pytest.mark.include_capsule
+def test_positive_service_list(sat_maintain):
+    """List satellite services using satellite-maintain service subcommand
+
+    :id: aa878d16-5a22-11ed-849d-cb7ac12cece2
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain service list
+
+    :expectedresults: Satellite services should be listed
+    """
+    result = sat_maintain.cli.Service.list()
+    logger.info(result.stdout)
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
```

Comment:
Do we have an ambition to check what particular services are listed and what is their status in this test case?